sources/api/netdog/src/interface_id.rs

```diff
@@ -0,0 +1,377 @@
+//! The interface_id module contains the definition of a valid network interface name and MAC
```

Comment:
Not sure why git decided this was a new file, rather than a giant update and rename from `interface_name`. :shrug:

Basically everything is new down to `InterfaceName`, and then the only changes beyond that are the additional unit tests.